cli/azd/pkg/project/service_target_aks.go

```diff
@@ -578,3 +575,40 @@ func (t *aksTarget) setK8sContext(ctx context.Context, serviceConfig *ServiceCon
 
 	return nil
 }
+
+// resolveClusterName attempts to resolve the cluster name from the following sources:
+// 1. The 'AZD_AKS_CLUSTER' environment variable
+// 2. The 'resourceName' property in the azure.yaml (Can use expandable string as well)
+// 3. The 'resourceName' property passed the target resource
+func (t *aksTarget) resolveClusterName(
+	serviceConfig *ServiceConfig,
+	targetResource *environment.TargetResource,
+) (string, error) {
+	// Resolve cluster name
+	var clusterName string
+	envVarClusterName, _ := t.env.LookupEnv(environment.AksClusterEnvVarName)
+	yamlClusterName, _ := serviceConfig.ResourceName.Envsubst(t.env.Getenv)
+
+	clusterNameResolutions := []string{
+		envVarClusterName,
+		yamlClusterName,
+		targetResource.ResourceName(),
+	}
+
+	for _, resourceName := range clusterNameResolutions {
+		if resourceName != "" {
+			clusterName = resourceName
+			break
+		}
+	}
```

Comment:
same as https://github.com/Azure/azure-dev/pull/3251/files#r1470356049.

a loop seems extra verbose for what it does